### PR TITLE
Sets up attachments to be destroyed for records set to be incinerated.

### LINKED ISSUE #18: When a record is incinerated, ensure that its associated S3 object (if any) is also deleted
@shreyansh-g24 _a 

@chiraggshah is already working on a POC for this. Sync with him.

### PR DESCRIPTION
**Pull request by shreyansh-g24**
FIXES #18

### CODE REVIEW

**Comment by shreyansh-g24** [comment]
@unnitallman _a Please check

**Comment by unnitallman** [comment]
@shreyansh-g24 Looks good. @chiraggshah _a Please take a look. 

cc: @neerajdotname

File: lib/org_incineration.rb [hunk 1]
```diff
@@ -67,12 +71,24 @@ def incinerate!(_model, model_hash)
...
         count = result.respond_to?(:count) ? result.count : result
       end
   end
 
+  def destroy_attachments_for(records, model)
+    unless model.reflect_on_all_associations.select { |m| m.class_name == "ActiveStorage::Attachment" }.present?
+      return
+    end
+
+    ActiveStorage::Attachment.where(record_id: records.pluck(:id)).destroy_all
```
Comment (chiraggshah) [inline lib/org_incineration.rb:89]: If we are not doing anything in the unless block, then an `if` condition would be easier to read.

```ruby
if model.reflect_on_all_associations.select { |m| m.class_name == "ActiveStorage::Attachment" }.present?
  ActiveStorage::Attachment.where(record_id: records.pluck(:id)).destroy_all
end
```

OR

```ruby
if model.reflect_on_all_associations.any?{ |m| m.class_name == "ActiveStorage::Attachment" }
  ActiveStorage::Attachment.where(record_id: records.pluck(:id)).destroy_all
end
```

**Review by chiraggshah — COMMENTED** [review]
@shreyansh-g24 _a : Looks good. A minor nit.

**Pull request merged**